components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt

```diff
@@ -324,6 +325,44 @@ class SystemEngineView @JvmOverloads constructor(
             session?.internalNotifyObservers { onContentPermissionRequest(SystemPermissionRequest(request)) }
         }
 
+        override fun onJsAlert(view: WebView?, url: String?, message: String?, result: JsResult): Boolean {
+            session?.apply {
+
+                val safeUrl = Uri.parse(
+                    if (url.isNullOrBlank()) {
+                        currentUrl
+                    } else {
+                        if (url.contains("about")) currentUrl else url
+                    }
+                )
```

Comment:
I do not really understand what this is doing. Why the "about" check? This block may need a comment :)

Reply:
Is this not an expected behavior ?

Reply:
I'm getting the `url` from `onJsAlert`. At least in this context `session` doesn't expose an `url`
![image](https://user-images.githubusercontent.com/773158/50651033-22f82980-0f50-11e9-890f-760464d660b5.png)